api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go

```diff
@@ -388,8 +391,54 @@ func (i *K8sTargetProvider) removeService(ctx context.Context, namespace string,
 			}
 		}
 	}
+
+	waitErr := i.pollCheck(ctx, true, func() error {
```

Comment:
It is better we introduce wait option. And only when wait in solution.components == true, we will do wait option. The similar logic is in helm target provider.

Reply:
added